gitlab/client.py

```diff
@@ -16,6 +18,14 @@
 import gitlab.exceptions
 from gitlab import utils
 
+from .clients.httpx_client import Httpx_Client
+from .clients.requests_client import Requests_Client
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
```

Comment:
If we do this, it also needs to be a conditional dependency in setup.py when python =< 3.7.

Reply:
OK. Worth adding a functional test which checks that. I've implemented something similar in a different project where I used a container to install the package and use it as the environment under test.

Reply:
IMO, it is in the scope of this PR.  This is how a maintainer can get confidence that the contributed change is not breaking something that worked before. Will work on it.

Reply:
@nejch I added a new workflow which installs the package and test whether `httpx` package can be imported or not. Not sure if this covers what you mentioned in your first comment. WDYT?